components/channel_header/index.ts

```diff
@@ -120,4 +124,4 @@ const mapDispatchToProps = (dispatch) => ({
     }, dispatch),
 });
 
-export default withRouter(connect(makeMapStateToProps, mapDispatchToProps)(ChannelHeader));
+export default withRouter<any, any>(connect(makeMapStateToProps, mapDispatchToProps)(ChannelHeader));
```

Comment:
I couldn't make `index` type checker happy without passing `any` to `withRouter`

Reply:
What were the errors? It looks like that shouldn't be needed based on [this file](https://github.com/mattermost/mattermost-webapp/blob/master/components/admin_console/index.ts), but maybe there's something different about how we defined the other props of the component

Reply:
Here is the error, which is not helpful at all (at least to me).

```
Argument of type 'ConnectedComponent<FC<WithIntlProps<Props>> & { WrappedComponent: ComponentType<Props>; }, Omit<WithIntlProps<Props>, "teamId" | ... 22 more ... | "isFileAttachmentsEnabled">>' is not assignable to parameter of type 'ComponentType<RouteComponentProps<any, StaticContext, unknown>>'.
  Type 'ConnectedComponent<FC<WithIntlProps<Props>> & { WrappedComponent: ComponentType<Props>; }, Omit<WithIntlProps<Props>, "teamId" | ... 22 more ... | "isFileAttachmentsEnabled">>' is not assignable to type 'FunctionComponent<RouteComponentProps<any, StaticContext, unknown>>'.
    Types of parameters 'props' and 'props' are incompatible.
      Type 'PropsWithChildren<RouteComponentProps<any, StaticContext, unknown>>' has no properties in common with type 'Omit<WithIntlProps<Props>, "teamId" | "currentUser" | "channel" | "memberCount" | "channelMember" | "dmUser" | "gmMembers" | "isFavorite" | "isReadOnly" | ... 14 more ... | "isFileAttachmentsEnabled">'.
```

I also found another and very similar example [here](https://github.com/mattermost/mattermost-webapp/blob/8a020cbed365385894c9c6b15eb6a536c5c067e8/components/global_header/center_controls/user_guide_dropdown/index.ts)

I also followed the same approach to use `ConnectedProps` like this but I am receiving the same error 
```
const connector = connect(makeMapStateToProps, mapDispatchToProps);

export type PropsFromRedux = ConnectedProps<typeof connector>;

export default withRouter(connector(ChannelHeader));
```

Reply:
Let's leave it as-is then. It must be something interfering between `useIntl`, `withRouter`, and our non-standard redux types. It would be nice if it works, but I'm not too concerned about it.